core/arch/arm/mm/core_mmu_lpae.c

```diff
@@ -817,19 +882,70 @@ void core_mmu_get_user_va_range(vaddr_t *base, size_t *size)
 {
 	assert(user_va_idx != -1);
 
+	/*
+	 * Base address is determined by base page table level granularity
+	 * (can be 0 or 1), while address space size is always of level 1.
+	 */
 	if (base)
-		*base = (vaddr_t)user_va_idx << L1_XLAT_ADDRESS_SHIFT;
+		*base = (vaddr_t)user_va_idx << BASE_XLAT_ADDRESS_SHIFT;
 	if (size)
 		*size = BIT64(L1_XLAT_ADDRESS_SHIFT);
 }
 
+static uint64_t *core_mmu_get_user_mapping_entry(struct mmu_partition *prtn,
+						 unsigned int base_idx)
+{
+	uint64_t *tbl;
+	unsigned int n;
+#if (CORE_MMU_BASE_TABLE_LEVEL == 0)
+	uintptr_t ntbl;
+#endif
+
+	assert(prtn);
+	assert(base_idx < NUM_BASE_TABLES);
+	assert(user_va_idx != -1);
+
+	tbl = prtn->base_tables[base_idx][get_core_pos()];
+	n = user_va_idx;
+
+	/*
+	 * If base level is 0, then user mapping entry is entry 0 in the level 1
+	 * table pointed by user_va_idx.
+	 */
+#if (CORE_MMU_BASE_TABLE_LEVEL == 0)
+	assert((tbl[n] & DESC_ENTRY_TYPE_MASK) == TABLE_DESC);
+
+	/* Copy bits 47:12 from tbl[n] to ntbl */
+	ntbl = tbl[n] & GENMASK_64(47, 12);
+
+#ifdef CFG_VIRTUALIZATION
+	if (prtn == &default_partition)
+		tbl = phys_to_virt(ntbl, MEM_AREA_TEE_RAM_RW_DATA);
+	else
+		tbl = phys_to_virt(ntbl, MEM_AREA_SEC_RAM_OVERALL);
+#else
+	tbl = phys_to_virt(ntbl, MEM_AREA_TEE_RAM_RW_DATA);
+#endif
+
+	n = 0;
+
+	assert(tbl);
+#endif
+
+	return &tbl[n];
+}
+
 bool core_mmu_user_mapping_is_active(void)
 {
 	bool ret;
 	uint32_t exceptions = thread_mask_exceptions(THREAD_EXCP_ALL);
+	uint64_t *entry;
```

Comment:
Please initialize all local variables with a value. Any value is OK as long as it isn't garbage from the stack.
This applies for the other function below that you've modified in this patch.